lib/spack/spack/patch.py

```diff
@@ -194,9 +194,11 @@ def fetch(self, stage):
         fetcher = fs.URLFetchStrategy(self.url, fetch_digest)
         mirror = os.path.join(
             os.path.dirname(stage.mirror_path),
+            'mirror-stage',
```

Comment:
This stemmed from testing (so mock environments) where the call to `expand_archive` for the patch would fail with "Already staged" because `self.expanded` (i.e., `self.source_path`) already existed.

Reply:
FYI. I'll have to revisit the previously failing tests since including path in the constructor is the source of the `mpich` build failure.  In other words, I'm removing these changes.

Reply:
Removed